ui/src/terraui/build.sh

```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env bash
+
+# http://redsymbol.net/articles/unofficial-bash-strict-mode/
+set -euo pipefail
+set -v
+
+# The date specified here must be early-enough to contain the specified hash.
+git clone --shallow-since='2022-11-29T00:00' \
```

Comment:
this is not idempotent

```
$ yarn build-terra-deps
yarn run v1.22.19
$ cd src/terraui; ./build.sh

# The date specified here must be early-enough to contain the specified hash.
git clone --shallow-since='2022-11-29T00:00' \
  https://github.com/DataBiosphere/terra-ui.git \
  .repo
fatal: destination path '.repo' already exists and is not an empty directory.
```